test/Microsoft.Dnx.Testing.CommonUtils/DnxSdk/Dnu.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Dnx.Testing
+{
+    public class Dnu
+    {
+        private readonly string _sdkPath;
+
+        public Dnu(string sdkPath)
+        {
+            _sdkPath = sdkPath;
+        }
+
+        public ExecResult Publish(
+            string projectPath,
+            string outputPath,
+            bool noSource = false,
+            string additionalArguments = null,
+            Action<Dictionary<string, string>> envSetup = null)
+        {
+            var sb = new StringBuilder();
+            sb.Append("publish ");
+            sb.Append($@"""{projectPath}""");
+            if (noSource)
+            {
+                sb.Append(" --no-source");
+            }
+
+            sb.Append($@" --out ""{outputPath}""");
+            sb.Append($" {additionalArguments}");
+
+            return Execute(sb.ToString(), envSetup);
+        }
+
+        public ExecResult Restore(
+            string restoreDir,
+            string packagesDir = null,
+            IEnumerable<string> feeds = null)
+        {
+            var sb = new StringBuilder();
+            sb.Append("restore");
```

Comment:
You can do some of these appends in one line. I think it more clear: `sb.Append($"restore \"{restoreDir}\")`